lib/spack/llnl/util/filesystem.py

```diff
@@ -414,48 +417,84 @@ def get_filetype(path_name):
     return output.strip()
 
 
+def chgrp_if_not_world_writable(path, group):
+    """chgrp path to group if path is not world writable"""
+    mode = os.stat(path).st_mode
+    if mode & stat.S_IWOTH:
```

Comment:
From the function name it seems to me that you want to `chgrp` if _path_ is **NOT** writable, but from the check I would say the opposite.